app.rb

```diff
@@ -95,4 +97,106 @@ def list_rentals
       puts "Date: #{e.date}, Book: #{e.book.title} by #{e.book.author}" if e.person.id == id
     end
   end
+
+  # ------------------------ #
+  # Save and Load operations #
+  # ------------------------ #
+  def save_data
+    save_books
+    save_people
+    save_rentals
+  end
+
+  def load_data
+    @books = load_books || []
+    @people = load_people || []
+    @rentals = load_rentals || []
+  end
+
+  def save_books
+    # File.write('./data/books.json', JSON.generate(@books))
+    # File.write('./data/books.json', Marshal.dump(@books))
+
+    arr = []
+    @books.each do |e|
+      arr.push(e.receive_item)
+    end
+    File.write('./data/books.json', JSON.generate(arr))
+  end
+
+  def load_books
+    # JSON.parse(File.read('./data/books.json')) if File.exist?('./data/books.json')
+    # Marshal.load(File.read('./data/books.json')) if File.exist?('./data/books.json')
+
+    return unless File.exist?('./data/books.json')
+
+    JSON.parse(File.read('./data/books.json')).map do |e|
+      id = e['id']
+      title = e['title']
+      author = e['author']
+      Book.new(title, author, id)
+    end
+  end
+
+  def save_people
+    # File.write('./data/people.json', JSON.generate(@people))
+    # File.write('./data/people.json', Marshal.dump(@people))
+
+    arr = []
+    @people.each do |e|
+      arr.push(e.receive_item)
+    end
+    File.write('./data/people.json', JSON.generate(arr))
+  end
+
+  def load_people
+    # JSON.parse(File.read('./data/people.json')) if File.exist?('./data/people.json')
+    # Marshal.load(File.read('./data/people.json')) if File.exist?('./data/people.json')
+
+    return unless File.exist?('./data/people.json')
+
+    JSON.parse(File.read('./data/people.json')).map do |e|
+      id = e['id'].to_i
+      people_class = e['class']
+      name = e['name']
+      age = e['age']
+      if people_class == 'student'
+        classroom = e['classroom']
+        parent_permission = e['parent_permission']
+        Student.new(classroom, age, name, parent_permission, id)
+      else
+        specialization = e['specialization']
+        Teacher.new(age, name, specialization, id)
+      end
+    end
+  end
+
+  def save_rentals
+    # File.write('./data/rentals.json', JSON.generate(@rentals))
+    # File.write('./data/rentals.json', Marshal.dump(@rentals))
+
+    arr = []
+    @rentals.each do |e|
+      arr.push(e.receive_item)
+    end
+    File.write('./data/rentals.json', JSON.generate(arr))
+  end
+
+  def load_rentals
+    # JSON.parse(File.read('./data/rentals.json')) if File.exist?('./data/rentals.json')
+    # Marshal.load(File.read('./data/rentals.json')) if File.exist?('./data/rentals.json')
```

Comment:
- [ ] Please remove the commented out code, it makes your code looks unprofessional.